cognite/client/_api/diagrams.py

```diff
@@ -78,30 +85,81 @@ def detect(
         min_tokens: int = 2,
         file_ids: Union[int, Sequence[int]] = None,
         file_external_ids: Union[str, Sequence[str]] = None,
-    ) -> DiagramDetectResults:
+        multiple_jobs: Optional[bool] = False,
+    ) -> Union[DiagramDetectResults, DetectJobBundle]:
+
         """Detect entities in a PNID.
         The results are not written to CDF.
         Note: All users on this CDF subscription with assets read-all and files read-all capabilities in the project,
         are able to access the data sent to this endpoint.
 
         Args:
-            entities (Sequence[Union[dict,CogniteResource]]): List of entities to detect
+            entities (Sequence[Union[dict, CogniteResource]]): List of entities to detect
             search_field (str): If entities is a list of dictionaries, this is the key to the values to detect in the PnId
             partial_match (bool): Allow for a partial match (e.g. missing prefix).
             min_tokens (int): Minimal number of tokens a match must be based on
             file_ids (Sequence[int]): ID of the files, should already be uploaded in the same tenant.
             file_external_ids (Sequence[str]): File external ids.
+            multiple_jobs (Optional[bool]): Enables you to publish multiple jobs. The method will return a DetectJobBundle
         Returns:
-            DiagramDetectResults: Resulting queued job. Note that .result property of this job will block waiting for results."""
-
+            DiagramDetectResults: Resulting queued job. Note that the .result property of this job will block waiting for results.
+            or
+            DetectJobBundle: This object will be able to handle multiple jobs. Note that the .result will block waiting for results.
+        Examples:
+                >>> from cognite.client import CogniteClient
+                >>> client = CogniteClient()
+                >>> retrieved_model = client.diagrams.detect(
+                    entities=[{"userDefinedField": "21PT1017","ignoredField": "AA11"}],
+                    search_field="userDefinedField",
+                    partial_match=True,
+                    min_tokens=2,
+                    file_ids=[101],
+                    file_external_ids=["Test1"],
+                )
+        """
+        items = self._process_file_ids(file_ids, file_external_ids)
         entities = [
             entity.dump(camel_case=True) if isinstance(entity, CogniteResource) else entity for entity in entities
         ]
 
+        if multiple_jobs:
+            # Limit
+            num_new_jobs = ceil(len(items) / DETECT_API_FILE_LIMIT)
+            if num_new_jobs > DETECT_API_STATUS_JOB_LIMIT:
+                raise ValueError(
+                    f"Number of jobs exceed limit of: '{DETECT_API_STATUS_JOB_LIMIT}'. Number of jobs: '{num_new_jobs}'"
+                )
+
+            detect_job_manager = DetectJobManager.instance()
+            if detect_job_manager.has_active_job():
```

Comment:
This just ensures that not more than one `DetectJobBundle` is active at the same time. Not fully sure that this will never be a limitation. One could increase this number in the future to allow 5 bundles or so to run at the same time with minor changes.

Reply:
I know, so if you have a script that create jobs in two projects you to wait for the jobs in one project before you start the ones in the other. Probably not a problem in practice - unless some users actually do that - just mentioning that you can't do that now.

Reply:
Briefly discussed with @torbjornopheim : We will change the `DetectJobManager` such that it uses a `Dict`to keep track of active job bundles **on each project and cluster**
That way you can have noe ative JobBundle on each project and cluster you are working on and don't get collisions.
You would still be limited to exactly one JobBundle on one project.

Reply:
Dictionary is implemented in this PR